Bottom: ab61a57f579078a33440e17d5bd8f2e57ed9d4ec
Top:    a96a551f3c2c73b2bfda2d9b970b709f162e8542
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-15 21:10:38 +0000

Refresh of play-with-perspective

---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 911099f..a2b1d2a 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -894,10 +894,12 @@ hidgl_fill_pcb_polygon (PolygonType *poly, const BoxType *clip_box)
 
   glPushAttrib (GL_STENCIL_BUFFER_BIT);                   // Save the write mask etc.. for final restore
   glPushAttrib (GL_STENCIL_BUFFER_BIT |                   // Resave the stencil write-mask etc.., and
-                GL_COLOR_BUFFER_BIT);                     // the colour buffer write mask etc.. for part way restore
+                GL_COLOR_BUFFER_BIT |                     // the colour buffer write mask etc.., and
+                GL_DEPTH_BUFFER_BIT);                     // the depth write-mask etc.. for part way restore
   glStencilMask (stencil_bit);                            // Only write to our stencil bit
   glStencilFunc (GL_ALWAYS, stencil_bit, stencil_bit);    // Always pass stencil test, ref value is our bit
   glColorMask (0, 0, 0, 0);                               // Disable writting in color buffer
+  glDepthMask (GL_FALSE);
 
   /* It will already be setup like this (so avoid prodding the state-machine):
    * glStencilOp (GL_KEEP, GL_KEEP, GL_REPLACE); // Stencil pass => replace stencil value
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index a48751b..ca42744 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -2462,7 +2462,7 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
   if (1) {
     GLfloat global_ambient[] = {0.0f, 0.0f, 0.0f, 1.0f};
     glLightModelfv (GL_LIGHT_MODEL_AMBIENT, global_ambient);
-    glLightModeli (GL_LIGHT_MODEL_LOCAL_VIEWER, GL_TRUE);
+    glLightModeli (GL_LIGHT_MODEL_LOCAL_VIEWER, GL_FALSE);
   }
   if (1) {
     GLfloat diffuse[] =  {0.3, 0.3, 0.3, 1.0};
